x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -813,4 +823,17 @@ export class AlertsClient<
   public isUsingDataStreams(): boolean {
     return this._isUsingDataStreams;
   }
+
+  private throwIfHasClusterBlockException(response: BulkResponse) {
```

Comment:
inside `resolveAlertConflicts`, we have some additional checks to sanitize the the bulk error response and then we iterate over the response items already. WDYT of moving this check in there? https://github.com/elastic/kibana/blob/e529a899bcfbb28e2d490642fc4d29ee522debf8/x-pack/plugins/alerting/server/alerts_client/lib/alert_conflict_resolver.ts#L297

Reply:
Yes I know I copied the code from there.
I put the code there first but returning one of the reasons became a problem.
And had to pass the result to multiple functions back to use in the taskRunner.
it is cleaner this way.